NOMENCLATURE.md

```diff
@@ -16,102 +16,102 @@ To begin, we'll define the concept of a _cluster_.
 registration, session management, and notification of task assignment.
 
 ### Worker
 
-A _Worker_ is a complete engine joined to a _cluster_. It receives and executes
-_tasks_ while reporting on their status. _Tasks_ include definitions of
-container runtimes.
+A _worker_ is a complete _Engine_ joined to a _cluster_. It receives and executes
+_tasks_ while reporting on their status.
 
 A worker's _agent_ coordinates the receipt of task assignments and ensures status
-is correctly reported to a _dispatcher_.
+is correctly reported to the _dispatcher_.
 
 #### Engine
 
-The _engine_ is shorthand for the _Docker Engine_. It runs containers
-distributed via _tasks_.
+The _Engine_ is shorthand for the _Docker Engine_. It runs containers
+distributed via the _scheduler_ -> _dispatcher_ -> _agent_ pipeline.
 
 #### Agent
 
 The _agent_ coordinates the dispatch of work for a _worker_. The _agent_
 maintains a connection to the _dispatcher_, waiting for the current set of
-tasks assigned to the node. Assigned tasks are then dispatched to the engine.
+tasks assigned to the node. Assigned tasks are then dispatched to the Engine.
 The agent notifies the _dispatcher_ of the current state of assigned tasks.
 
-This is roughly analagous to an entertainment agent. It ensures the worker has
-the correct set of work and let's others know what the worker is doing.
+This is roughly analagous to a real life talent agent who ensures the worker
+has the correct set of _tasks_ and lets others know what the worker is doing.
 
-While we refer to cluster engines as a "worker", the term _agent_ encompasses
+While we refer to cluster Engines as a "worker", the term _agent_ encompasses
 only the component of a worker that communicates with the dispatcher.
 
 ## Objects
 
-An _Object_ is any configuration component accessed as a top-level component.
-These typically include a set of API to introspect objects and manipulate them
-through a _Spec_. 
+An _object_ is any configuration component accessed at the top-level. These
+typically include a set of APIs to inspect the objects and manipulate them
+through a _spec_. 
 
-_Objects_ are typically broken up into a _Spec_ component and a set of fields
-to keep track of the implementation of the Spec. The _Spec_ represents the
-users intent. When a user wants to modify an object, only the Spec portion is
-provided. When an object flows through the system, the Spec portion is left
+_objects_ are typically broken up into a _spec_ component and a set of fields
+to keep track of the implementation of the _spec_. The _spec_ represents the
+users intent. When a user wants to modify an object, only the spec portion is
+provided. When an object flows through the system, the spec portion is left
 untouched by all cluster components.
 
-Examples of objects include _Service_, _Task_, _Network_ and _Volume_.
+Examples of _objects_ include _service_, _Task_, _Network_ and _Volume_.
```

Comment:
Might be correct uppercase but with backticks. i think these might be the names of objects in the code.